test/xbgpu/test_engine.py

```diff
@@ -669,13 +668,15 @@ async def _send_data(
                 corrprod_results[i][j] = ig_recv["xeng_raw"].value
 
         # TODO: NGC-1172 The tweaks to process beam data below rely on
-        # `batch_indices` to be contiguous. The check below is temporary until
-        # the BPipeline is able to handle missing data.
-        assert sorted(batch_indices) == (
+        # `batch_indices` to be contiguous and for the zeroth (and minimum)
+        # value to be a multiple of `HEAPS_PER_FENGINE_PER_CHUNK`. The check
+        # below is temporary until the BPipeline is able to handle missing
+        # data.
```

Comment:
You've documented the extra assumption (zeroth element is a multiple) but haven't asserted it.